processor/src/errors.rs

```diff
@@ -21,10 +26,95 @@ pub enum ExecutionError {
+            }
+            ExecutionError::AdviceTapeReadFailed(step) => {
+                write!(fmt, "Advice tape read failed at step {step}")
+            }
+            ExecutionError::CodeBlockNotFound(digest) => {
+                write!(
+                    fmt,
+                    "Execution of call instruction failed, function body {digest:?} not found"
+                )
+            }
+            ExecutionError::CallerNotInSyscall => {
+                write!(
+                    fmt,
+                    "Instruction `caller` used outside of kernel context, this is not supported"
+                )
+            }
+            ExecutionError::DivideByZero(clk) => {
+                write!(fmt, "Division by zero at clock {clk}")
+            }
+            ExecutionError::DuplicateAdviceKey(key) => {
+                write!(fmt, "Insertion into advice map failed because {key:?} already exists")
+            }
+            ExecutionError::FailedAssertion(clk) => {
+                write!(fmt, "Assertion failed at clock {clk}")
+            }
+            ExecutionError::UninitializedMemoryAddress(address) => {
+                write!(fmt, "Ext2INTT referenced unintialized memory at address {address}")
+            }
+            ExecutionError::InvalidFmpValue(old, new) => {
+                write!(fmt, "Updating FMP register failed, from {old} to {new}, because {new} is outside of {FMP_MIN}..{FMP_MAX}")
+            }
+            ExecutionError::NttDomainSizeTooSmall(v) => {
+                write!(fmt, "Input domain evaluation length can not be zero {v}")
+            }
+            ExecutionError::NttDomainSizeNotPowerof2(v) => {
+                write!(fmt, "Input domain evaluation must be a power of two, got {v}")
+            }
+            ExecutionError::InvalidStackDepthOnReturn(depth) => {
+                write!(fmt, "Maximum supported stack depth when returning from call is {STACK_TOP_SIZE}, got {depth}")
+            }
+            ExecutionError::InterpolationResultSizeTooBig(output_len, input_len) => {
+                write!(
+                    fmt,
+                    "Output length {output_len} after polynomial evaluation must be smaller than inputlen {input_len}"
+                )
+            }
+            ExecutionError::NotBinaryValue(v) => {
+                write!(fmt, "Execution failed, was expecting a true/false value, got ({v})")
+            }
+            ExecutionError::NotU32Value(v) => {
+                write!(fmt, "Execution failed because op u32assert2 failed, value is not u32 ({v})")
+            }
+            ExecutionError::ProverError(error) => {
+                write!(fmt, "Prover failed {error}")
+            }
+            ExecutionError::SyscallTargetNotInKernel(proc) => {
+                write!(
+                    fmt,
+                    "Procedure call failed because proc {proc:?} does not exist in the kernel ROM"
```

Comment:
nit: I would change this to: "Syscall failed: procedure with root {proc} was not found in the kernel".